src/server/dflycmd.cc

```diff
@@ -764,18 +711,15 @@ std::map<uint32_t, LSN> DflyCmd::ReplicationLags() const {
   shard_set->RunBriefInParallel([&shard_lags, this](EngineShard* shard) {
     auto& lags = shard_lags[shard->shard_id()];
     for (const auto& info : replica_infos_) {
-      if (info.second->state.load() != SyncState::STABLE_SYNC) {
-        lags[info.first] = std::numeric_limits<LSN>::max();
-        continue;
+      const ReplicaInfo* replica = info.second.get();
+      if (shard->journal()) {
+        int64_t lag = shard->journal()->GetLsn() - replica->flows[shard->shard_id()].last_acked_lsn;
+        lags[info.first] = lag;
```

Comment:
I'd suggest adding a `DCHECK(!replica.mu.try_lock())` to verify it's locked when the code around changes

Reply:
it's not locked though and we do not access any of the fields that can be accessed by multiple threads.

Reply:
Ok, agree. ReplicaInfo::flows is pre-sized and last_acked_lsn is only touched during stable sync from it's respective thread